organizationmembership/client_test.go

```diff
@@ -0,0 +1,207 @@
+"role":"%s",
+"organization":{"id":"%s"},
+"public_user_data":{"user_id":"%s"}
+}`,
+				id, role, organizationID, userID)),
+			Method: http.MethodPatch,
+			Path:   "/v1/organizations/" + organizationID + "/memberships/" + userID,
+		},
+	}
+	client := NewClient(config)
+	membership, err := client.Update(context.Background(), &UpdateParams{
+		Role:           clerk.String(role),
+		OrganizationID: organizationID,
+		UserID:         userID,
+	})
+	require.NoError(t, err)
+	require.Equal(t, id, membership.ID)
+	require.Equal(t, role, membership.Role)
+	require.Equal(t, organizationID, membership.Organization.ID)
+	require.Equal(t, userID, membership.PublicUserData.UserID)
+}
+
+func TestOrganizationMembershipClientUpdate_Error(t *testing.T) {
+	t.Parallel()
+	config := &ClientConfig{}
+	config.HTTPClient = &http.Client{
+		Transport: &clerktest.RoundTripper{
+			T:      t,
+			Status: http.StatusBadRequest,
+			Out: json.RawMessage(`{
+  "errors":[{
+		"code":"update-error-code"
+	}],
+	"clerk_trace_id":"update-trace-id"
+}`),
+		},
+	}
+	client := NewClient(config)
+	_, err := client.Update(context.Background(), &UpdateParams{})
+	require.Error(t, err)
+	apiErr, ok := err.(*clerk.APIErrorResponse)
+	require.True(t, ok)
+	require.Equal(t, "update-trace-id", apiErr.TraceID)
+	require.Equal(t, 1, len(apiErr.Errors))
+	require.Equal(t, "update-error-code", apiErr.Errors[0].Code)
+}
+
+func TestOrganizationMembershipClientDelete(t *testing.T) {
+	t.Parallel()
+	id := "orgmem_123"
+	organizationID := "org_123"
+	userID := "user_123"
+	config := &ClientConfig{}
+	config.HTTPClient = &http.Client{
+		Transport: &clerktest.RoundTripper{
+			T: t,
+			Out: json.RawMessage(fmt.Sprintf(`{
+"id":"%s",
+"organization":{"id":"%s"},
+"public_user_data":{"user_id":"%s"}
+}`,
+				id, organizationID, userID)),
+			Method: http.MethodDelete,
+			Path:   "/v1/organizations/" + organizationID + "/memberships/" + userID,
+		},
+	}
+	client := NewClient(config)
+	membership, err := client.Delete(context.Background(), &DeleteParams{
+		UserID:         userID,
+		OrganizationID: organizationID,
+	})
+	require.NoError(t, err)
+	require.Equal(t, id, membership.ID)
+	require.Equal(t, organizationID, membership.Organization.ID)
+	require.Equal(t, userID, membership.PublicUserData.UserID)
+}
+
+func TestOrganizationMembershipClientList(t *testing.T) {
+	t.Parallel()
+	id := "orgmem_123"
+	organizationID := "org_123"
+	userID := "user_123"
+	config := &ClientConfig{}
+	config.HTTPClient = &http.Client{
+		Transport: &clerktest.RoundTripper{
+			T: t,
+			Out: json.RawMessage(fmt.Sprintf(`{
+"data": [{
+	"id":"%s",
+	"organization":{"id":"%s"},
+	"public_user_data":{"user_id":"%s"}
+}],
+"total_count": 1
+}`,
+				id, organizationID, userID)),
+			Method: http.MethodGet,
+			Path:   "/v1/organizations/" + organizationID + "/memberships",
+			Query: &url.Values{
+				"limit":    []string{"1"},
+				"offset":   []string{"2"},
+				"role":     []string{"admin", "member"},
```

Comment:
❓ Ah here's a case for an array, right? https://github.com/clerk/clerk-sdk-go/pull/231/files#r1482950532